File: u2/python/DemoClient.py
import socket
import sys

def main():
    # connect to server
    with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as s: #create a TCP socket
        s.connect(('127.0.0.1', 12345))
        line = 1
        print('Input something to send to server or empty line to end')
				
        while line:
            # Ctrl-C to terminate the loop
            line = sys.stdin.readline().strip()
            if line: #send out the message only when the input is not empty
                s.send((line + "\n").encode('utf-8')) # send out the message in utf-8 format. I have added a '\n' to handle the reading text in server side for Java version
                reply = s.recv(1024).strip() # get the server reply
                print (reply.decode('utf-8'))

if __name__ == "__main__":
    main()

File: u2/java/DemoClient.java
import java.io.*;
import java.net.*;

public class DemoClient{
	
	public static void main(String args[]){
		try{
			String _str = "";	
			
			Socket _socket = new Socket("127.0.0.1", 12345); //Prepare TCP socket and connect to server
			BufferedReader _in = new BufferedReader(new InputStreamReader(_socket.getInputStream())); //Get the input stream, for getting message from server to client.
			PrintWriter _out = new PrintWriter(_socket.getOutputStream(), true); //Get the output stream, for sending message from client to server.
			BufferedReader _usrInput = new BufferedReader(new InputStreamReader(System.in)); //Prepare the input stream for reading user input from standard input stream

			System.out.println("Type something or press Ctrl-C to end");
			while((_str = _usrInput.readLine()) != null){ //read the user input
				_out.println(_str);//send out to server
				System.out.println(_in.readLine()); //get the message from server and print it out
			}
			
			/* Close all those socket and stream stuffs */
			_usrInput.close();
			_in.close();
			_out.close();
			_socket.close();
			System.out.println("Connection end");
		}catch(IOException _ioe){
			_ioe.printStackTrace();
		}catch(Exception _e){
			_e.printStackTrace();
		}
	}
}